### ISSUE TITLE
cmd/compile: inline static init cause compile time error

### ISSUE THREAD

**Issue by cuonglm** [golang/go#58439]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.20 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env

</pre></details>

### What did you do?

```
package main

var x = f(-1)
var y = f(64)

func f(x int) int {
	return 1 << x
}

func main() {
	println(x, y)
}
```

### What did you expect to see?

Runtime error.

### What did you see instead?

Compile time error.

**Comment by cuonglm** [comment]
The fix for #58293 only deal with the integer overflow case.

**Comment by cuonglm** [comment]
This needs to be backport, too cc @randall77 @mdempsky @dr2chase 

There're a plenty of issue with inline static init happens, and although I think https://go-review.googlesource.com/c/go/+/466277 should fix them, maybe a better way to backport is disable inline static init for go1.20, then re-enable by default during go1.21 cycle.

**Comment by randall77** [comment]
I agree, probably the safest thing to do for 1.20 is just disable inline static init. We can fix for 1.21 properly.

**Comment by mdempsky** [comment]
Disabling the optimization for 1.20 seems right to me too, unfortunately.

**Comment by cuonglm** [comment]
@gopherbot please backport this issue to 1.20
